go/vt/vtgate/vtgate.go

```diff
@@ -209,6 +209,11 @@ var (
 		"VtgateApiRowsAffected",
 		"Rows affected by a write (DML) operation through the VTgate API",
 		[]string{"Operation", "Keyspace", "DbType"})
+
+	sqlTextCounts = stats.NewCountersWithMultiLabels(
+		"VtgateSQLTextCounts",
```

Comment:
How does `/metrics` render this one? If it turns out to be vtgate_s_q_l_text_counts, we'll want to rename it :)
In general, for any metrics changes, it is good to post screenshots or embedded text from both `/debug/vars` and `/metrics` into the PR description. That serves as a visual check for what we are emitting.

Reply:
OK, now I understand the intent of this metric, it won't work as implemented. Counters should only be used for monotonically increasing numbers, prometheus will not let you reduce the value of a "counter".
The right type of metric to use here is a Gauge, which can take any value, and change up or down. If I had read the linked issue properly, I would have realized this during the original review.
The name of the metric is also misleading, because it is not a "count" of some specific kind of sql text, it is instead the length in text of the query that is being processed, so the name needs to reflect that.

Reply:
`QueryTextProcessed` will in fact be a better name for the metric.

Reply:
The way I understand it, the metric would be more appropriately named `QueryProcessedTextLengths`